### PR TITLE
feat(Rich Text Editor): mention plugin improve

### PR DESCRIPTION
**Pull request by bojangles-m**
Mention Combobox item component update

### CODE REVIEW

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxItem.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React from 'react';
+import { ComboboxItemProps } from '@udecode/plate';
+import { MentionItemData, MentionableCategory } from '../types';
+import { IconPeople, IconPerson, IconSize, IconTarget } from '@foundation/Icon';
+
+type RenderAvatarProps = {
+    image: string | undefined;
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxItem.tsx:9]:
```suggestion
    image?: string;
```

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxItem.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            break;
+        case MentionableCategory.ALL:
+            avatar = <IconTarget size={IconSize.Size12} />;
+            break;
+        default:
+            avatar = <IconPerson size={IconSize.Size12} />;
+            break;
+    }
+
+    return image ? <img src={image} alt={`${id}-${category}-${text}`} /> : avatar;
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxItem.tsx:29]:
```suggestion
    if (image) {
        return <img src={image} alt={`${id}-${category}-${text}`} />
    }
    
    switch (category) {
        case MentionableCategory.GROUP:
            return <IconPeople size={IconSize.Size12} />;
        case MentionableCategory.ALL:
            return <IconTarget size={IconSize.Size12} />;
        default:
            return <IconPerson size={IconSize.Size12} />;
    }
```

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxItem.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+export const MentionComboboxItem = ({ item }: ComboboxItemProps<MentionItemData>): JSX.Element => {
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxItem.tsx:32]: Prefer `ReactElement` over `JSX.Element`
```suggestion
export const MentionComboboxItem = ({ item }: ComboboxItemProps<MentionItemData>): ReactElement => {
```

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionInputMarkupElement/MentionInputMarkupElementNode.tsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React from 'react';
+import { MentionInputElement, MentionInputElementProps, Value } from '@udecode/plate';
+import { MentionInputMarkupElementStyles } from './MentionInputMarkupElementStyles';
+
+export const MentionInputMarkupElementNode = (props: MentionInputElementProps<Value>) => (
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/Plugins/MentionPlugin/MentionInputMarkupElement/MentionInputMarkupElementNode.tsx:7]: Don't forget `: ReactElement` so the return is ensured to the JSX and not something else
```suggestion
export const MentionInputMarkupElementNode = (props: MentionInputElementProps<Value>): ReactElement => (
```

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxStyles.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            padding-top: 0.5rem;
+            padding-bottom: 0.5rem;
+            z-index: 500;
+            border-radius: var(--radius);
+            border: 1px solid var(--line-color-strong);
+            box-shadow: var(--shadow-matrix);
+            background: var(--base-color);
+        `,
+    ],
+    item: [...item],
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxStyles.ts:28]: Why spreading?
```suggestion
    item,
```

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxItem.tsx (hunk 1 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxItem.tsx:9]: Updated

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxItem.tsx (hunk 2 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxItem.tsx:29]: Updated

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxItem.tsx (hunk 3 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxItem.tsx:32]: Updated

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionInputMarkupElement/MentionInputMarkupElementNode.tsx (hunk 4 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/Plugins/MentionPlugin/MentionInputMarkupElement/MentionInputMarkupElementNode.tsx:7]: Updated

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxStyles.ts (hunk 5 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/Plugins/MentionPlugin/MentionCombobox/MentionComboboxStyles.ts:28]: Thnx, yes of course. Updated

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**